salt/files/master.d/f_defaults.conf

```diff
@@ -746,14 +746,21 @@ rosters:
 {%- do default_keys.append('master_tops') %}
 master_tops:
 {%- for master in cfg_master['master_tops'] -%}
-  {%- if cfg_master['master_tops'][master] is string %}
-  {{ master }}: {{ cfg_master['master_tops'][master] }}
-  {%- else %}
-  {{ master}}:
+  {%- if cfg_master['master_tops'][master] is mapping %}
+  {{ master }}:
     {%- for parameter in cfg_master['master_tops'][master] %}
     {{ parameter }}: {{ cfg_master['master_tops'][master][parameter] }}
     {%- endfor -%}
-  {%- endif -%}
+  {%- elif cfg_master['master_tops'][master] is string %}
```

Comment:
@0xf10e , I also though like this initially, but it wasn't correct, because string is iterable, so the behaviour was not the expect one.
I think that the correct configuration can be created only if the string is treated before other iterable items.